back/engines/commercial/id_vienna_saml/app/lib/id_vienna_saml/citizen_saml_omniauth.rb

```diff
@@ -36,6 +36,11 @@ def profile_to_user_attrs(auth)
       }
     end
 
+    def profile_to_user_attrs_for_update(auth)
```

Comment:
It's a very good point. It comes from my first attempt to simplify it https://github.com/CitizenLabDotCo/citizenlab/pull/7354/commits/fbfb0cab31c6c1d96527608f79a34baef17592d5#diff-449def09404dcd17c09d1bc691616fbe82fe7a609a4945ace787fa1de24d6c57R40-R44

But then I realized that first and last names are always set the first time a user registers and later it can be updated by SSO only if the user sets their names to empty strings. And in this case, we should probably respect their choice.

Reply:
So, I completely removed names from `updateable_user_attrs` for Vienna Citizens.

Reply:
> if the user sets their names to empty strings

I'm a bit surprised this is allowed though.

Reply:
:man_shrugging: